pystac/item.py

```diff
@@ -352,6 +352,36 @@ def get_collection(self) -> Optional[Collection]:
         else:
             return cast(Collection, collection_link.resolve_stac_object().target)
 
+    def set_derived_from(self, item: Optional[Item]) -> Item:
```

Comment:
I feel like this needs to take an iterable for the case where `itemA` is derived from `itemB` and `itemC`

Reply:
Ok cool cool, I'll update this with `add_derived_from(self, *items)`. I'm on the fence about how `remove_derived_from` should work or if it should exist at all. As I see it it could either remove all the `derived_from` links which can also be done with `item.remove_links("derived_from")` or it could remove one or more items from the `derived_from` links. I guess that is the more useful one so that it can be like the opposite of `add_`

Reply:
I think you are correct that `remove_derived_from` is `self.remove_links("derived_from")`.  Doesn't seem needed, iff you stick with `set_derived_from(self, *items)`, since that will wipe the `derived_from`s out for you.  

If you move to `add_derived_from` (which linguistically implies it won't remove anything), then we might want to doc that `item.remove_links("derived_from")` can be used for removal in the docstring, if not adding the `remove_derived_from` convenience function.  Not sure about the case of removing a subset of the `derived_from` links.  Feels like that moves into the territory of `item.set_derived_from(link for link in item.get_derived_from() if conditional(link))`.

Reply:
Ok I have remove_derived_from accepting a single item_id. See what you think